tests/test_TFNetworkRecLayer.py

```diff
@@ -3598,6 +3598,21 @@ def test_reclayer_optimize_out_access_split():
     other_subnet_layers={"split": {"class": "split", "from": ["data:source"], "size_splits": [5, 8]}})
 
 
+def test_reclayer_optimize_out_slice_nd():
+  check_reclayer_optimize_out(
+    {"class": "linear", "activation": None, "from": ["encoder_reduced"]},
+    from_="position",
+    other_subnet_layers={
+      "window": {"class": "slice_nd", "from": "base:encoder", "start": "data:source", "size": None, "min_size": 1, "is_output_layer": True},
+      "encoder_reduced": {"class": "reduce", "mode": "sum", "axis": "T", "from": ["base:encoder"], "is_output_layer": True}},
+    shared_base_net={
+      "encoder": {"class": "copy", "from": ["data"], "is_output_layer": True},
+      "position": {"class": "eval", "from": ["encoder"], "is_output_layer": True,
+                "eval": "tf.zeros(tf.shape(source(0, enforce_batch_major=True, auto_convert=False))[:-1], dtype=tf.dtypes.int32)",
+                "out_type": {"batch_dim_axis": 0, "time_dim_axis": 1, "shape": (None,),
+                             "sparse": True, "dtype": "int32", "dim": None}}})
```

Comment:
Do not indent it like that. Always use a new line + 2 space indent instead of these hanging indents.